### PR TITLE
Add help messages for most of the deprecated stuff in ZScript.

### PR DESCRIPTION
**Pull request by nashmuhandes**
Some of these might be more verbose than they need to be. Will definitely need to be developer-verified to ensure the information is accurate.

### CODE REVIEW

**Comment by Cacodemon345** [comment]
I believe support for formatting deprecated warning would be more useful here (in most cases here), if it was supported, like:
```deprecated("2.3", "Use %s instead",function)```
Where function is not a string, it instead references a function, whose name would be printed.

**Comment by coelckers** [comment]
Looking good, but let's have another person review it too, to make sure I didn't overlook anything.

File: wadsrc/static/zscript/base.zs [hunk 1]
```diff
@@ -740,7 +740,7 @@ struct LevelLocals native
 	native static void MakeScreenShot();
 	native static void MakeAutoSave();
 	native void WorldDone();
-    deprecated("3.8") static void RemoveAllBots(bool fromlist) { /* intentionally left as no-op. */ }
+    deprecated("3.8", "This function does nothing") static void RemoveAllBots(bool fromlist) { /* intentionally left as no-op. */ }
```
Comment (alexey-lysiuk) [inline wadsrc/static/zscript/base.zs]: As you are modifying this line, please change leading spaces to a tab.

File: wadsrc/static/zscript/compatibility.zs [hunk 2]
```diff
@@ -27,7 +27,7 @@ extend class Actor
 	//
 	//==========================================================================
 
-	deprecated("3.7") bool CheckClass(class<Actor> checkclass, int ptr_select = AAPTR_DEFAULT, bool match_superclass = false)
+	deprecated("3.7", "Use GetClass() instead") bool CheckClass(class<Actor> checkclass, int ptr_select = AAPTR_DEFAULT, bool match_superclass = false)
```
Comment (alexey-lysiuk) [inline wadsrc/static/zscript/compatibility.zs]: "Use GetClass() with a cast or 'is' operator instead"

**Comment by alexey-lysiuk** [comment]
Merged manually in 7a141f3aa3b67b5b1d326f5d9b3904da1b65f847.

**Comment by nashmuhandes** [comment]
> > Add help messages for most of the deprecated stuff
> 
> What's remaining without messages?

I deliberately skipped these as I felt these were too obscure and weren't exposed to users.

// At the end of actor.zs
	// Internal functions
	deprecated("2.3") private native int __decorate_internal_int__(int i);
	deprecated("2.3") private native bool __decorate_internal_bool__(bool b);
	deprecated("2.3") private native double __decorate_internal_float__(double f);

// inventory.zs
	deprecated("3.7") private int ItemFlags;

// weapon.zs
	deprecated("3.7") uint WeaponFlags;		// not to be used directly.
	deprecated("3.7") int MinAmmo1, MinAmmo2;	// not used anywhere and thus deprecated.

// loadsavemenu.zs
	deprecated("4.0") void DrawSaveComment(Font font, int cr, int x, int y, int scalefactor)

// menu.zs
	deprecated("4.0") static void DrawConText (int color, int x, int y, String str)

// textentermenu.zs
	deprecated("3.8") static TextEnterMenu Open(Menu parent, String textbuffer, int maxlen, int sizemode, bool showgrid = false, bool allowcolors = false)

// statscreen.zs
	deprecated("3.8") TextureID mPatch;
	deprecated("3.8") int DrawPatchText(int y, PatchInfo pinfo, String stringname)

**Comment by nashmuhandes** [comment]
> Merged manually in [7a141f3](https://github.com/coelckers/gzdoom/commit/7a141f3aa3b67b5b1d326f5d9b3904da1b65f847).

Sorry for the late response. I was literally fixing everything you commented a few minutes before you did the manual merge.

**Pull request closed without merging**